packages/svelte-query-persist-client/tests/PersistQueryClientProvider.svelte.test.ts

```diff
@@ -129,45 +126,37 @@ describe('PersistQueryClientProvider', () => {
       props: {
         queryClient,
         persistOptions: { persister },
-        states: statesStore,
+        states,
       },
     })
 
     await waitFor(() => rendered.getByText('fetchStatus: idle'))
     await waitFor(() => rendered.getByText('hydrated'))
     await waitFor(() => rendered.getByText('fetched'))
 
-    const states = get(statesStore)
-
-    expect(states).toHaveLength(4)
+    expect(states.value).toHaveLength(3)
 
-    expect(states[0]).toMatchObject({
+    expect(states.value[0]).toMatchObject({
       status: 'pending',
       fetchStatus: 'idle',
       data: undefined,
     })
 
-    expect(states[1]).toMatchObject({
+    expect(states.value[1]).toMatchObject({
       status: 'success',
       fetchStatus: 'fetching',
       data: 'hydrated',
     })
 
-    expect(states[2]).toMatchObject({
```

Comment:
It looks like this deleted state is identical to the previous state, so likely a bug in the svelte v4 project.